### PR TITLE
Only allow approved operators to register nodes, expose nodes by operator

### PR DESCRIPTION
**Pull request by pfives**
https://testnets.llama.xyz/orgs/river/base-sepolia/actions/116

### CODE REVIEW

**Review by mechanical-turk — APPROVED** [review]

File: contracts/src/base/registry/facets/checker/EntitlementChecker.sol [hunk 1]
```diff
@@ -153,6 +160,27 @@ contract EntitlementChecker is IEntitlementChecker, Facet {
...
+    address operator
+  ) external view returns (address[] memory) {
+    EntitlementCheckerStorage.Layout storage layout = EntitlementCheckerStorage
+      .layout();
+    uint256 nodeCount = layout.nodes.length();
+    address[] memory nodes = new address[](nodeCount);
+    for (uint256 i = 0; i < nodeCount; i++) {
+      nodes[i] = layout.nodes.at(i);
+      if (layout.operatorByNode[nodes[i]] == operator) {
+        nodes[i] = layout.nodes.at(i);
```
Comment (clemire) [inline contracts/src/base/registry/facets/checker/EntitlementChecker.sol:178]: This is the same as line 176? Were you trying to build a packed array of nodes here? I think that's one loop to count and one loop to assign.

File: contracts/test/crosschain/EntitlementChecker.t.sol [hunk 2]
```diff
@@ -121,4 +135,26 @@ contract EntitlementCheckerTest is BaseSetup, IEntitlementCheckerBase {
...
     entitlementChecker.getRandomNodes(26);
   }
+
+  // =============================================================
+  //                        Nodes by Operator
+  // =============================================================
+
+  function test_getNodesByOperator() external {
+    for (uint256 i = 0; i < operators.length; i++) {
+      uint256 totalNodes = 0;
```
Comment (clemire) [inline contracts/test/crosschain/EntitlementChecker.t.sol:145]: Can you rewrite this test case to catch any errors pointed out above?

**Review by jterzis — APPROVED** [review]

File: contracts/src/base/registry/facets/checker/EntitlementChecker.sol (hunk 1 above)
Comment (jterzis) [reply contracts/src/base/registry/facets/checker/EntitlementChecker.sol:178]: https://github.com/river-build/river/pull/171

File: contracts/test/crosschain/EntitlementChecker.t.sol (hunk 2 above)
Comment (clemire) [reply contracts/test/crosschain/EntitlementChecker.t.sol:145]: For this test case, I was thinking specifically, validate that these are the nodes registered by a specific operator. Do you think these changes would have caught the error that was fixed above where all nodes were included?

**Pull request merged**